test/helpers/account.js

```diff
@@ -4,10 +4,10 @@ const { impersonateAccount, setBalance } = require('@nomicfoundation/hardhat-net
 // Hardhat default balance
 const DEFAULT_BALANCE = 10000n * ethers.WeiPerEther;
 
-async function impersonate(account, balance = DEFAULT_BALANCE) {
-  await impersonateAccount(account);
-  await setBalance(account, balance);
-}
+const impersonate = (account, balance = DEFAULT_BALANCE) =>
+  impersonateAccount(account)
+    .then(() => setBalance(account, balance))
+    .then(() => ethers.getSigner(account));
```

Comment:
This change proposes that "impersonate" returns an ethers signer. That way, you no longer need to instanciate the signer after you called impersonate.